### PR TITLE
Lesson size functionality cleanup

### LINKED ISSUE #10236: Wrong modal is displayed when toggling off the 'Visible to learners' toggle
## Observed behavior
When I go to Coach > Plan > Lesson and I attempt to toggle off the 'Visible to learners' toggle I see the 'Make lesson visible' modal, while since I want to make it invisible the modal should be 'Make lesson not visible'. This is functioning correctly at Coach > Reports > Lessons

## Expected behavior
The correct modal should be displayed

## Steps to reproduce the issue
1. Install the the [following build ](https://buildkite.com/learningequality/kolibri-debian/builds/6281) 
2.  Coach > Plan > Lesson and I try to toggle off the 'Visible to learners' toggle
3. Observe the text of the displayed modal

## Video

https://user-images.githubusercontent.com/79847249/225036061-2af71d4e-b6f7-490b-b92b-1e367977ceb8.mp4

## Usage Details
Windows 10, Ubuntu 22  - Chrome

### LINKED ISSUE #10500: Coach > Reports - The total size of lessons visible to learners is not displayed
## Observed behavior
While testing https://github.com/learningequality/kolibri/pull/10452 I noticed that the total size of lessons visible to learners is not displayed at Coach > Reports. The value is incorrectly displayed as 0 and after refreshing the page as: --
This could be related to the issue described in https://github.com/learningequality/kolibri/issues/10483

## Expected behavior
The correct value for the lesson size should be displayed

## Steps to reproduce the issue
1. Install the the following [Kolibri 0.16a12 build ](https://github.com/learningequality/kolibri/suites/12297334558/artifacts/651689066) 
2. Go to Coach > Reports > Lesson
3. Observe the value for the total size of lessons visible to learners and the Size value in the table

## Video

https://user-images.githubusercontent.com/79847249/232793908-a953de64-bc66-4111-9153-cc1a3cb17ce3.mp4

## Usage Details
Ubuntu - Chrome, Firefox

### LINKED ISSUE #10501: Coach > Plan - Problems with calculating the Total size of lessons visible to learners
## Observed behavior
Observed while testing https://github.com/learningequality/kolibri/pull/10452 - the total size of lessons visible to learners is not calculated correctly after I add more than a single lesson at Coach > Plan. See the video for details.

## Expected behavior
The correct value for the total lesson size should be displayed

## Steps to reproduce the issue
1. Install the the following [Kolibri 0.16a12 build ](https://github.com/learningequality/kolibri/suites/12297334558/artifacts/651689066) 
2. Go to Coach > Plan > Lesson
3. Add a single lesson with resources and observe the total size
4. Add a second lesson and note that the total size value is displayed as 'NaN B'. Also the lesson size value in the table for the first lesson is no longer correct.

## Video

https://user-images.githubusercontent.com/79847249/232797414-b29e6ad9-fc78-47c5-9734-3c0d3ede00f9.mp4

## Usage Details
Ubuntu - Chrome, Firefox

### LINKED ISSUE #10617: Toggle button not working properly with its props
## Observed behavior

There are toggle buttons to help the user to switch  between two different states but here in some parts, the toggle button is not working properly and it's not able to change its state 

Moreover, even its able to change its state but unable to display those changes in the frontend 

## Errors and logs

[Screencast from 02-05-23 06:15:09 PM IST.webm](https://user-images.githubusercontent.com/81693090/235673299-e3d75c21-fa6d-4f95-8bc5-75713e0fc29c.webm)

## Expected behavior

It should be able to toggle between visible to non-visible with the correct active display 

## User-facing consequences

Its quite difficult for the admin to identify whether it is visible to learners

## Steps to reproduce

1. Navigate towards Coach 
2. Navigate towards your class
3. Click on any particular lesson 
4. Toggle the button visible to learners

## Context

…

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Cleans up lesson size fetching to always pass the `classId` as the collection get parameter.
* Adds `id` filter to be able to fetch just the size information for a single lesson
* Updates vuex state storing of lesson sizes to be parallel to quiz sizes by storing it directly on the lesson data in the vuex modules
* Does some fly by cleanup of `forEach` to `for... of`
* Removes hardcoded `--` string and uses `KEmptyPlaceholder` for cases where a blank should be shown.
* Cleans up some residual active/is_active issues

## References
Fixes #10500 
Closes #10501 
Fixes #10236 
Fixes #10617 

## Reviewer guidance
#10500:
[Screencast from 05-18-2023 12:12:09 PM.webm](https://github.com/learningequality/kolibri/assets/1680573/1b9663d6-2b81-4083-a75a-29b6d10cccbd)

#10501:
[Screencast from 05-18-2023 12:13:20 PM.webm](https://github.com/learningequality/kolibri/assets/1680573/8230796a-0613-4c99-b8d6-0d008d6ef747)

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by pcenov — DISMISSED** [review]
Hi @rtibbles, looks line there is one more base to cover: in Coach > Plan > Lesson the Size value is always displayed as 'NaN B'. This is also valid for the modal window which is displayed when there's a LOD:

Video:

https://github.com/learningequality/kolibri/assets/79847249/81f6eb4d-21fc-4500-8c6f-96c2b4abb006



Screenshot:

File: kolibri/plugins/coach/assets/src/modules/classSummary/index.js [hunk 1]
```diff
@@ -453,33 +462,35 @@ export default {
...
     },
     fetchLessonsSizes(store, classId) {
-      return LessonResource.fetchLessonsSizes({ collection: classId })
-        .then(sizes => {
-          store.commit('SET_CLASS_LESSONS_SIZES', sizes);
-        })
-        .catch(error => {
-          return store.dispatch('handleApiError', error, { root: true });
-        });
+      if (Object.keys(store.state.lessonMap).length > 0) {
```
Comment (marcellamaki) [inline kolibri/plugins/coach/assets/src/modules/classSummary/index.js:472]: this is a much better place to put this condition!

File: kolibri/plugins/coach/assets/src/views/plan/LessonsRootPage/index.vue [hunk 2]
```diff
@@ -218,18 +218,18 @@
...
-          return size;
+        if (this.lessons && this.lessons.length) {
+          const sum = this.lessons
+            .filter(
+              // only include visible lessons
+              lesson => lesson.is_active
+            )
+            .reduce((acc, lesson) => {
+              return acc + (lesson.size || 0);
+            }, 0);
```
Comment (marcellamaki) [inline kolibri/plugins/coach/assets/src/views/plan/LessonsRootPage/index.vue:229]: this is a nice cleanup

**Review by marcellamaki — COMMENTED** [review]
Code read through looks good - the cleanup makes it much, much easier to read. I can do one more pass after the code is updated for the code that Peter mentioned.

**Comment by rtibbles** [comment]
But `NaN B` is my favourite grandmother!

I'll clean this up - must be one more edge case lurking.

**Comment by rtibbles** [comment]
@pcenov when you confirm the final fix here, can you also verify that it fixes this issue? #10617

**Comment by rtibbles** [comment]
This should now be displaying properly, as shown here:
image

**Comment by pcenov** [comment]
Hi @rtibbles, I confirm the final fix here. I've reopened one [previously reported issue](https://github.com/learningequality/kolibri/issues/10236) which I was able to identify while regression testing everything.

**Comment by rtibbles** [comment]
I will do the final fix for this issue within this PR too.

**Comment by rtibbles** [comment]
This now also fixes #10236:
image

**Comment by radinamatic** [comment]
Regression mentioned in #10236 is fixed, but if you look closely at the lesson sizes right after the modal is closed, you can still see a brief flash of NaN... 

https://github.com/learningequality/kolibri/assets/1457929/4a26e475-b496-4523-aa8e-d42e2846658a

lesson-visibility2 mp4 - VLC media player_001

**Comment by rtibbles** [comment]
I deliberately didn't fix these loading issues to avoid scope creep into this issue: https://github.com/learningequality/kolibri/issues/10729

**Review by radinamatic — APPROVED** [review]
Manual QA passed, good to go! 🎉 💯 :shipit:

**Pull request merged**